### ISSUE TITLE
No InteractionMatrix possible for HCP crystallography.

### ISSUE THREAD

**Issue by daria-mesbah** [thelfer/tfel#450]
Hello,

I have tried to develop a crystal plasticity model using `FiniteStrainSingleCrystal` brick with a HCP crystallography (including prismatic, basal, pyramidal1A slip families), as in the attached mfront file. The command `mfront-query --interaction-matrix-structure HCP_FiniteStrain.mfront` returns the exact number of independent coefficients (20). However, unlike to FCC structure described in the documentation, the command line `mfront-query --interaction-matrix HCP_FiniteStrain.mfront` returns an error as follows:
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  SlipSystemsDescription::InteractionMatrixStructure::getRank: internal error (unsupported slip system type)
|Abort
```
It seems that HCP slip systems are not considered to build-up the interaction matrix structure. Fixing this issue would allow me to use the interaction matrix in the hardening laws with the `Integrator` brick. 

Thanks in advance for your kind help,
Daria MESBAH

**Comment by thelfer** [comment]
Hi Daria,

Would you give me a minimal example to reproduce the issue ?

Thomas

**Comment by daria-mesbah** [comment]
Hi Thomas,

Thanks your your quick reply. You will find attached a pretty basic `mfront` file of HCP crystallography, only intended for `mfront-query` purposes. (Sorry for the file format, it seems that github does not allow mfront file transfer.)

Daria

[HCP_FiniteStrain.txt](https://github.com/thelfer/tfel/files/13390631/HCP_FiniteStrain.txt)

**Comment by thelfer** [comment]
Well, it seems working:

~~~~
$ mfront-query --interaction-matrix HCP_FiniteStrain.mfront
| 0 1 1 2 3 2 4 5 5 5 5 4 |
| 1 0 1 2 2 3 5 4 5 5 4 5 |
| 1 1 0 3 2 2 5 5 4 4 5 5 |
| 6 6 7 8 9 9 10 10 11 11 10 10 |
| 7 6 6 9 8 9 11 10 10 10 10 11 |
| 6 7 6 9 9 8 10 11 10 10 11 10 |
| 12 13 13 14 15 14 16 17 17 18 18 19 |
| 13 12 13 14 14 15 17 16 18 17 19 18 |
| 13 13 12 15 14 14 17 18 16 19 17 18 |
| 13 13 12 15 14 14 18 17 19 16 18 17 |
| 13 12 13 14 14 15 18 19 17 18 16 17 |
| 12 13 13 14 15 14 19 18 18 17 17 16 |
~~~~

**Comment by thelfer** [comment]
What version are you using ?

**Comment by daria-mesbah** [comment]
Here is the version I am using
```
tfel
Version : 4.0.0-dev
Compiled with  on  - 
Please submit bug at tfel-contact@cea.fr
```
Which version would you recommend to use instead ?

**Comment by thelfer** [comment]
@daria-mesbah The current version of TFEL is 4.1. Version 4.2 will be out in a week. I would recommend to use any of the ` rliv-4.1` or `master`  branches.

If you need to stick with the 4.0.x branch, the latest version is 4.0.1. I don't know if this version fixes your issue, but you can use the `rliv-4.0` branch to get the latest of the  4.0.x branch, i.e. version 4.0.x with all the bug fixes.

**Comment by daria-mesbah** [comment]
Thank you for your kind help. I will upgrade the `mfront` version.
